src/airmar/airmar_receiver.py

```diff
@@ -1,57 +1,99 @@
-        self.port = port
-        self.processor = AirmarProcessor(broadcaster=broadcaster)
+        self.uart_pin = read_pin_config(mock_bbio=mock_bbio)
+        self.port = read_port_config(mock_port=mock_port)
+        self.logger = logger
+        self.processor = AirmarProcessor(
+            broadcaster=broadcaster, parser=self.parser)
 
     def start(self):
         """ Sets up uart pin and open port to start listening. 
         Enables sentences specified by ids field to airmar serial port."""
+        # Setup port/pin
         self.uart_pin.setup()
-        # Close port before open needed during force quit.
-        self.port.close()
         self.port.open()
-        # Resumes sentence transmition
+
+        # Airmar specific sentence transmissions to airmar port.
+        # Resumes sentence transmission
         self.port.write(
             "{}".format(self.parser.power(resume=1)).encode(self.port.encoding))
-        # Disable all sentence transmitions first.
+        
+        # First, disable all transmissions
         self.port.write(
             "{}".format(self.parser.toggle(enable=0)).encode(self.port.encoding))
         toggles = self.parser.toggle(self.ids)
         for toggle in toggles:
-            # Enables sentences specified by config
+            # Second, enable specified transmissions in config
             self.port.write("{}".format(toggle).encode(self.port.encoding))
+        
         self.is_running = True
 
     def send_airmar_data(self):
-        """ Sends nmea sentence from serial port to processor to broadcast data """
-        data = self.parser.parse(self.port.read_line(terminator='\r\n'))
-        # sentence type can be determined by first element of data.
-        if data is not None:
+        """ Sends nmea sentence from serial port to processor to broadcast data 
+        
+        Side effects:
+        Updates logger with warnings and errors
+        """
+        sentence = self.port.read_line(terminator='\r\n')
+
+        mutex.acquire()
+        try:
+            data = self.parser.parse(sentence)
             self.processor.update_airmar_data(nmea=data)
 
+        except InvalidIDException:
+            self.logger.write_msg(pin_name=self.uart_pin.pin_name,
+            msg=r"WARNING Invalid SID: \"{}\"".format(sentence), 
+            rw_state="r")
+        except InvalidSentenceException:
+            self.logger.write_msg(pin_name=self.uart_pin.pin_name,
+            msg=r"WARNING Unable to parse \"{}\"".format(sentence), 
+            rw_state="r")
+        except UnsupportedIDException:
+            self.logger.write_msg(pin_name=self.uart_pin.pin_name,
+            msg=r"WARNING Unsupported ID: \"{}\"".format(sentence), 
+            rw_state="r")
+        except Exception as e:
+            self.logger.write_msg(pin_name=self.uart_pin.pin_name,
+            msg=r"ERROR Unhandled Exception\"{}\": \"{}\"".format(e, sentence), 
+            rw_state="r")
+
+        mutex.release()
```

Comment:
you only need to hold the mutex while you're worried about other threads impacting the mutable object... I think you're safe to release it at the end of the try block